qiskit_experiments/test/mock_iq_helpers.py

```diff
@@ -0,0 +1,393 @@
+         circuit. The function returns a list of dictionaries, each containing output binary strings and
+          their probabilities.
+
+        Examples:
+
+            **1 qubit circuit - excited state**
+
+            In this experiment, we want to bring a qubit to its excited state and measure it.
+            The circuit:
+                         ┌───┐┌─┐
+                      q: ┤ X ├┤M├
+                         └───┘└╥┘
+                    c: 1/══════╩═
+                               0
+
+            The function that calculates the probability for this circuit, doesn't need any
+            calculation_parameters. It will be as following:
+
+            .. code-block::
+
+                @staticmethod
+                def compute_probability() -> List[Dict[str, float]]:
+
+                    output_dict_list = []
+                    for circuit in circuits:
+                        probability_output_dict["1"] = 1
+                        probability_output_dict["0"] = 0
+                        output_dict_list.append(probability_output_dict)
+                    return output_dict_list
+
+            **3 qubit circuit**
+            In this experiment, we prepare a Bell state with the first and second qubit.
+            In addition, we will bring the third qubit to its excited state.
+            The circuit:
+                         ┌───┐     ┌─┐
+                    q_0: ┤ H ├──■──┤M├───
+                         └───┘┌─┴─┐└╥┘┌─┐
+                    q_1: ─────┤ X ├─╫─┤M├
+                         ┌───┐└┬─┬┘ ║ └╥┘
+                    q_2: ┤ X ├─┤M├──╫──╫─
+                         └───┘ └╥┘  ║  ║
+                    c: 3/═══════╩═══╩══╩═
+                                2   0  1
+
+            The backend has a feature, that the output can include only states with probabilities
+            greater than 0. The backend will assume that if an output string isn't in the probability
+            dictionary, it's probability 0.
+
+            .. code-block::
+
+                @staticmethod
+                def compute_probability_3q() -> List[Dict[str, float]]:
+
+                    output_dict_list = []
+                    for circuit in circuits:
+                        probability_output_dict["001"] = 0.5
+                        probability_output_dict["111"] = 0.5
+                        output_dict_list.append(probability_output_dict)
+                    return output_dict_list
+        """
+
+    # pylint: disable=unused-argument
+    def iq_phase(self, circuit: QuantumCircuit) -> float:
+        """Sub-classes can override this method to introduce a phase in the IQ plan.
+
+        This is needed, to test the resonator spectroscopy where the point in the IQ
+        plan has a frequency-dependent phase rotation.
+        """
+        return 0.0
+
+
+class MockIQDragHelper(MockIQExperimentHelper):
+    """functions needed for test_drag"""
```

Comment:
```suggestion
    """Functions needed for test_drag"""
```

Reply:
Convert `functions` to `Functions` in all classes below as well

Reply:
Changed as suggested.